src/cpp/fastdds/xtypes/dynamic_types/DynamicTypeImpl.cpp

```diff
@@ -0,0 +1,217 @@
+    return RETCODE_OK;
+}
+
+uint32_t DynamicTypeImpl::get_verbatim_text_count() noexcept
+{
+    return static_cast<uint32_t>(verbatim_.size());
+}
+
+ReturnCode_t DynamicTypeImpl::get_verbatim_text(
+        traits<VerbatimTextDescriptor>::ref_type& descriptor,
+        uint32_t idx) noexcept
+{
+    if (!descriptor || idx >= verbatim_.size())
+    {
+        return RETCODE_BAD_PARAMETER;
+    }
+
+    traits<VerbatimTextDescriptor>::narrow<VerbatimTextDescriptorImpl>(descriptor)->copy_from(verbatim_.at(idx));
+    return RETCODE_OK;
+}
+
+bool DynamicTypeImpl::equals(
+        traits<DynamicType>::ref_type other) noexcept
+{
+    bool ret_value = true;
+    auto impl = traits<DynamicType>::narrow<DynamicTypeImpl>(other);
+
+    ret_value &= type_descriptor_.equals(impl->type_descriptor_);
+    if (ret_value)
+    {
+        ret_value &= annotation_.size() == impl->annotation_.size();
+        if (ret_value)
+        {
+            for (size_t count {0}; ret_value && count < annotation_.size(); ++count)
+            {
+                ret_value &= annotation_.at(count).equals(impl->annotation_.at(count));
+            }
+        }
+
+        ret_value &= member_.size() == impl->member_.size();
+        assert(TK_ANNOTATION == type_descriptor_.kind() ||
+                TK_BITMASK == type_descriptor_.kind() ||
+                TK_BITSET == type_descriptor_.kind() ||
+                TK_STRUCTURE == type_descriptor_.kind() ||
+                TK_UNION ==  type_descriptor_.kind() ||
+                0 == member_.size());
+        assert(TK_ANNOTATION == type_descriptor_.kind() ||
+                TK_BITMASK == impl->type_descriptor_.kind() ||
+                TK_BITSET == impl->type_descriptor_.kind() ||
+                TK_STRUCTURE == impl->type_descriptor_.kind() ||
+                TK_UNION ==  impl->type_descriptor_.kind() ||
+                0 == impl->member_.size());
+
+        assert(member_by_name_.size() == members_.size());
+        assert(impl->member_by_name_.size() == impl->members_.size());
+        ret_value &= member_by_name_.size() == impl->member_by_name_.size();
+        if (ret_value)
+        {
+            auto it = member_by_name_.begin();
+            auto impl_it = impl->member_by_name_.begin();
+
+            while (ret_value && member_by_name_.end() != it)
+            {
+                ret_value &= it->second->equals(impl_it->second);
+                ++it;
+                ++impl_it;
+            }
+        }
+
+        ret_value &= verbatim_.size() == impl->verbatim_.size();
+        if (ret_value)
+        {
+            for (size_t count {0}; ret_value && count < verbatim_.size(); ++count)
+            {
+                ret_value &= verbatim_.at(count).equals(impl->verbatim_.at(count));
+            }
+        }
+    }
+
+    return ret_value;
+}
+
+traits<DynamicType>::ref_type DynamicTypeImpl::_this ()
+{
+    return shared_from_this();
```

Comment:
<memory> header file is not being included.

Reply:
Done in c8b1745ba